vividus-engine/src/main/java/org/vividus/log/LoggingTableTransformerMonitor.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2019-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.log;
+
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jbehave.core.model.ExamplesTable.TableProperties;
+import org.jbehave.core.model.TableParsers;
+import org.jbehave.core.model.TableTransformerMonitor;
+import org.jbehave.core.model.TableTransformers.Formatting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LoggingTableTransformerMonitor implements TableTransformerMonitor
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoggingTableTransformerMonitor.class);
+
+    private static final Formatting FORMATTING_TRANSFORMER = new Formatting();
+    private static final int MAX_ROWS_LOGGING = 50;
+
+    private final TableParsers tableParsers;
+
+    public LoggingTableTransformerMonitor(TableParsers tableParsers)
+    {
+        this.tableParsers = tableParsers;
+    }
+
+    @Override
+    public void beforeTransformerApplying(String transformerName, TableProperties properties, String inputTable)
+    {
+        String loggerFormat = String
+                .format("Applying table transformer: {}%nTabular parameters: {}%nInput table body:%n{}");
+        logTransformerInfo(transformerName, properties, inputTable, loggerFormat);
+    }
+
+    @Override
+    public void afterTransformerApplying(String transformerName, TableProperties properties, String outputTable)
+    {
+        String loggerFormat = String
+                .format("Applying table transformer: {}%nTabular parameters: {}%nOutput (transformed) table:%n{}");
+        logTransformerInfo(transformerName, properties, outputTable, loggerFormat);
+    }
+
+    private void logTransformerInfo(String transformerName, TableProperties tableProperties, String table,
+            String loggerFormat)
+    {
+        String propertiesForLogging = tableProperties.getPropertiesAsString().trim();
+        String tableForLogging = getExampleTableForLogging(table, tableProperties);
+
+        LOGGER.info(loggerFormat, transformerName, propertiesForLogging, tableForLogging);
```

Comment:
there is lots of stuff going on with strings, but I think we should take into account a low chance that the info level might be disabled, so in this case, all these computations should not be performed